presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,357 @@
+        }
+
+        public int getEnd()
+        {
+            return end;
+        }
+
+        public FieldType getFieldType()
+        {
+            return fieldType;
+        }
+
+        public int getLength()
+        {
+            return end - start;
+        }
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        for (int i = 0; i < columnMappings.get(currentColumnIndex).getLength(); i++) {
+            buffer.put((byte) 0);
+        }
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        buffer.putLong(value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        buffer.putInt(value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        buffer.putShort(value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        buffer.put(value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        buffer.putDouble(value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        buffer.putFloat(value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        buffer.put((byte) (value ? 1 : 0));
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        byte[] valueBytes = value.getBytes(StandardCharsets.UTF_8);
+        checkArgument(valueBytes.length == columnMappings.get(currentColumnIndex).getLength(), format(
+                "length '%s' of message '%s' for column '%s' does not equal expected length '%s'",
+                valueBytes.length,
+                value,
+                columnHandles.get(currentColumnIndex).getName(),
+                columnMappings.get(currentColumnIndex).getLength()));
+        buffer.put(valueBytes, 0, valueBytes.length);
+    }
+
+    @Override
+    protected void appendByteBuffer(ByteBuffer value)
+    {
+        byte[] valueBytes = value.array();
+        checkArgument(valueBytes.length == columnMappings.get(currentColumnIndex).getLength(), format(
+                "length '%s' of message for column '%s' does not equal expected length '%s'",
+                valueBytes.length,
+                columnHandles.get(currentColumnIndex).getName(),
+                columnMappings.get(currentColumnIndex).getLength()));
+        buffer.put(valueBytes, 0, valueBytes.length);
+    }
+
+    @Override
+    public byte[] toByteArray()
+    {
+        // make sure entire row has been updated with new values
+        checkArgument(currentColumnIndex == columnHandles.size(), format("Missing %d columns", columnHandles.size() - currentColumnIndex + 1));
+
+        resetColumnIndex(); // reset currentColumnIndex to prepare for next row
+        buffer.clear(); // reset buffer position to prepare for next row
```

Comment:
first `clear()` and then `array()`. I am supprised it works. I think we should just mark intention to `clear` here and do actual clearing on next append.

Reply:
`clear()` doesn't actually get rid of any of the data in the backing array, it just resets the position and limit

Reply:
Yeah. You are right. I was expecting that this is just imlementation detail but actually it is documented behaviour. Can you make the comment by `buffer.clear()` more explicit so it is obvious that we know what we are doing, and avoid confustion of next generation of people reading this code :)

Reply:
yeah for sure